File: finalFrontendApplication/src/app/features/companies/controllers/company-controller.ts
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Company } from "../entities/Company";
import { HttpClient } from "@angular/common/http";




@Injectable({ providedIn: "root" })
export class CompanyController {

    private url = "http://localhost:8080/companies";


    constructor(private http: HttpClient) {}


    // get all companies
    getCompanies(): Observable<Company[]> {
        return this.http.get<Company[]>(this.url);
    }

    // get company by id
    getCompanyById(id: string): Observable<Company> {
        return this.http.get<Company>(`${this.url}/${id}`);
    }

    // create company
    createCompany(company: Company): Observable<Company> {
        return this.http.post<Company>(this.url, company);
    }

    // update company
    updateCompany(id: string, company: Company): Observable<Company> {
        return this.http.put<Company>(`${this.url}/${id}`, company);
    }


    // delete company
    deleteCompany(id: string): Observable<void> {
        return this.http.delete<void>(`${this.url}/${id}`);
    }





}

File: Android/app/src/main/java/com/bivolaris/myapplication/controller/CompanyController.java
package com.bivolaris.myapplication.controller;

import com.bivolaris.myapplication.model.Company;
import com.bivolaris.myapplication.model.CreateCompanyRequest;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface CompanyController {
    @GET("companies")
    Call<List<Company>> getCompanies();

    @GET("companies/{id}")
    Call<Company> getCompanyById(@Path("id") String id);

    @POST("companies")
    Call<Void> createCompany(@Body CreateCompanyRequest company);


    @PUT("companies/{id}")
    Call<Company> updateCompany(@Path("id") String id, @Body Company company);

    @DELETE("companies/{id}")
    Call<Void> deleteCompany(@Path("id") String id);


}


